libr/asm/arch/arm/winedbg/be_arm.c

```diff
@@ -466,26 +442,14 @@ static ut16 thumb_disasm_pushpop(struct winedbg_arm_insn *arminsn, ut16 inst) {
 static ut16 thumb_disasm_blocktrans(struct winedbg_arm_insn *arminsn, ut16 inst) {
 	short load = (inst >> 11) & 0x01;
 	short i;
-	short last;
+	bool first = true;
 
-	for (i=7;i>=0;i--) {
-		if ((inst>>i) & 1) {
-			break;
-		}
-	}
-	last = i;
-
-	//TODO PJ
 	arminsn->str_asm = r_str_appendf (arminsn->str_asm, "%s %s!, {", load ? "ldmia" : "stmia", tbl_regs[(inst >> 8) & 0x07]);
 
-	for (i=0;i<=7;i++) {
-		if ((inst>>i) & 1) {
-			if (i == last) {
-				arminsn->str_asm = r_str_appendf (arminsn->str_asm, "%s", tbl_regs[i]);
-			}
-			else {
-				arminsn->str_asm = r_str_appendf (arminsn->str_asm, "%s, ", tbl_regs[i]);
-			}
+	for (i = 0; i <= 7; i++, inst >>= 1) {
+		if (inst & 1) {
+			arminsn->str_asm = r_str_appendf (arminsn->str_asm, "%s%s", first ? "" : ", ", tbl_regs[i]);
```

Comment:
Rstrget in this first var

Reply:
> Rstrget in this first var

https://github.com/radareorg/radare2/blob/44755fe08d53b7426bbd33c2b5a0e0796d5cc343/libr/util/str.c#L823-L825
Not applicable.